lib/puppet/parser/functions/foreman.rb

```diff
@@ -36,6 +36,7 @@
 #
 # Happy Foreman API-ing!
 
+require "pry"
```

Comment:
i had tried to put 25 seconds for timeout, somehow it`s does`nt solve problem. after installing pry from first run its work perfectly. Section about  timeout  is less important.